### PR TITLE
feat: fee saved display

### PR DESCRIPTION
**Pull request by ChefJerry**
---

## PR-Codex overview
The focus of this PR is to update translations in the localization config, add DAI token support for multiple chains, and implement fee savings calculation in the Swap UI.

### Detailed summary
- Updated translations in `translations.json`
- Added DAI token support for various chains in `common.ts`
- Implemented fee savings calculation in Swap UI
- Added `useFeeSaved` hook for fee calculation
- Updated `AdvancedSwapDetails.tsx` and `useFeeSaved.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by ChefMomota — APPROVED** [review]

File: apps/web/src/views/Swap/components/AdvancedSwapDetails.tsx [hunk 1]
```diff
@@ -57,6 +59,30 @@ export const TradeSummary = memo(function TradeSummary({
...
+                </>
+              }
+              ml="4px"
+              placement="top"
+            />
+          </RowFixed>
+          <RowFixed>
+            <Text fontSize="14px" color="success">
+              {t('Up to ')}
+              {formatAmount(feeSaved, 4)} {inputAmount?.currency?.symbol}
```
Comment (chefjackson) [inline apps/web/src/views/Swap/components/AdvancedSwapDetails.tsx:81]: Let's use `NumberDisplay` from internal widgets

File: apps/web/src/views/Swap/hooks/useFeeSaved.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { DAI, USDC, USDT } from '@pancakeswap/tokens'
+import { useMemo } from 'react'
+
+const MAJOR_DEFINED_STABLE_COINS = {
+  [ChainId.ETHEREUM]: [USDC[ChainId.ETHEREUM], DAI[ChainId.ETHEREUM], USDT[ChainId.ETHEREUM]],
+  [ChainId.ARBITRUM_ONE]: [USDC[ChainId.ARBITRUM_ONE]],
+  [ChainId.ARBITRUM_GOERLI]: [USDC[ChainId.ARBITRUM_GOERLI]],
+  [ChainId.POLYGON_ZKEVM]: [USDC[ChainId.POLYGON_ZKEVM], DAI[ChainId.POLYGON_ZKEVM]],
+  [ChainId.POLYGON_ZKEVM_TESTNET]: [USDC[ChainId.POLYGON_ZKEVM_TESTNET]],
+  [ChainId.BSC]: [USDC[ChainId.BSC]],
```
Comment (chefjackson) [inline apps/web/src/views/Swap/hooks/useFeeSaved.ts:11]: how about `usdt`? I think swapping between `usdt` and `usdc` doesn't charge for interface fee as well?

File: apps/web/src/views/Swap/hooks/useFeeSaved.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const { chainId } = inputAmount.currency
+
+    const majorDefinedStableCoins = MAJOR_DEFINED_STABLE_COINS[chainId] ?? []
+    const zeroFee = [inputAmount.currency, outputAmount.currency].every((currency) =>
+      majorDefinedStableCoins.some((coin) => coin.equals(currency)),
+    )
+
+    if (zeroFee) return null
+
+    return inputAmount.multiply(25).divide(10000)
```
Comment (chefjackson) [inline apps/web/src/views/Swap/hooks/useFeeSaved.ts:29]: The fee should be paid in output token?

File: apps/web/src/views/Swap/hooks/useFeeSaved.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    const zeroFee = [inputAmount.currency, outputAmount.currency].every((currency) =>
+      majorDefinedStableCoins.some((coin) => coin.equals(currency)),
+    )
+
+    if (zeroFee) return undefined
+
+    return outputAmount.multiply(25).divide(10000)
+  }, [inputAmount, outputAmount])
+
+  const feeSavedUsdValue = useStablecoinPriceAmount(inputAmount?.currency, Number(feeSavedAmount?.toExact()), {
```
Comment (chefjackson) [inline apps/web/src/views/Swap/hooks/useFeeSaved.ts:41]: Let's use USD price from wallet api. It has multiple data sources covered.

useCurrencyUsdPrice hook

**Review by chefjackson — APPROVED** [review]

**Pull request merged**